coffeecard/CoffeeCard.Library/Services/v2/ProductService.cs

```diff
@@ -119,13 +119,28 @@ public async Task<ChangedProductResponse> UpdateProduct(UpdateProductRequest cha
 
             await _context.SaveChangesAsync();
 
+            var existingUserGroups = _context.ProductUserGroups.Where(e => e.ProductId == changedProduct.Id);
```

Comment:
The `GetProductAsync` method already returns a Product where its ProductUserGroups are included, thus there is no reason to call the DB to find the ProductUserGroup. Instead you can just manipulate the property directly as done for the other properties above.

Reply:
Please also write a unit test for the above scenario, thanks!